Automattic/wp-calypso
# Move Badge component to @automattic/components

Related to #77564
* https://github.com/Automattic/wp-calypso/pull/78809

## Proposed Changes

* Copies the `Badge` component and related tests, examples, etc. to `@automattic/components` so we can use it in other packages (like Launchpad).
* References to Badge to the new package will be updated on https://github.com/Automattic/wp-calypso/pull/78812.
* The original PR(linked above) was reverted due to an error executing `yarn build-packages`. It would fail to find the `PropTypes` while building the packages:
https://github.com/Automattic/wp-calypso/blob/1a10a9a4892d77df834f8c0667ef6adc1df6e909/packages/components/src/badge/index.jsx#L9
  * More context: p1688481304807469-slack-C02DQP0FP  



## Testing Instructions



* Make sure all tests pass.
```bash
yarn run test-packages packages/components/src/badge/test/index.js
```
* Run `yarn build-packages` and make sure it works
* This alone should not break anything since it's making a copy of the files in question. :) 

## Pre-merge Checklist
